Framework/Interaction/Rest/Customer.php

```diff
@@ -0,0 +1,126 @@
+use Magento\Framework\DataObjectFactory;
+use Psr\Log\LoggerInterface;
+
+class Customer extends Rest implements RestCustomerInterface
+{
+    /**
+     * @var CustomerHelper
+     */
+    protected $customerHelper;
+
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * @param CustomerHelper    $customerHelper
+     * @param Config            $config
+     * @param LoggerInterface   $logger
+     * @param DataObjectFactory $dataObjectFactory
+     * @param ClientPool        $clientPool
+     */
+    public function __construct(
+        CustomerHelper $customerHelper,
+        Config $config,
+        LoggerInterface $logger,
+        DataObjectFactory $dataObjectFactory,
+        ClientPool $clientPool
+    )
+    {
+        parent::__construct($logger, $dataObjectFactory, $clientPool);
+
+        $this->customerHelper = $customerHelper;
+        $this->config = $config;
+    }
+
+    /**
+     * Perform REST request to get companies associated with the account
+     *
+     * @param DataObject      $request
+     * @param bool|null       $isProduction
+     * @param string|int|null $scopeId
+     * @param string          $scopeType
+     *
+     * @return DataObject[]
+     * @throws \ClassyLlama\AvaTax\Exception\AvataxConnectionException
+     */
+    public function getCertificatesList(
+        $request,
+        $isProduction = null,
+        $scopeId = null,
+        $scopeType = \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+    )
+    {
+        $client = $this->getClient($isProduction, $scopeId, $scopeType);
+
+        if ($request->getData('customer_code')) {
+            throw new \InvalidArgumentException('Must include a request with customer id');
+        }
+
+        $clientResult = $client->listCertificatesForCustomer(
+            $this->config->getCompanyId($scopeId, $scopeType),
+            $this->customerHelper->getCustomerCode($request->getData('customer_id'), null, $scopeId),
+            $request->getData('include'),
+            $request->getData('filter'),
+            $request->getData('top'),
+            $request->getData('skip'),
+            $request->getData('order_by')
+        );
+
+        $this->validateResult($clientResult, $request);
+
+        $certificates = $this->formatResult($clientResult)->getValue();
+
+        return $certificates;
+    }
+
+    /**
+     * Perform REST request to get companies associated with the account
+     *
+     * @param DataObject      $request
+     * @param bool|null       $isProduction
+     * @param string|int|null $scopeId
+     * @param string          $scopeType
+     *
+     * @return mixed
+     */
+    public function downloadCertificate(
+        $request,
+        $isProduction = null,
+        $scopeId = null,
+        $scopeType = \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+    )
+    {
+        $client = $this->getClient($isProduction, $scopeId, $scopeType);
+
+        return $client->downloadCertificateImage(
```

Comment:
@LordZardeck - Why are we not doing validateResult and formatResult here?

Reply:
Because this is an oclet stream, not json data that can be formatted